tools/StaticAnalysis/HelpAnalyzer/HelpAnalyzer.cs

```diff
@@ -265,30 +285,88 @@ public void Analyze(IEnumerable<string> scopes, IEnumerable<string> modulesToAna
+            script = $"Import-LocalizedData -BaseDirectory {parentDirectory} -FileName {psd1FileName} -BindingVariable ModuleMetadata; $ModuleMetadata.FunctionsToExport;";
+            cmdletResult = PowerShell.Create().AddScript(script).Invoke();
+            var functionCmdlets = cmdletResult.Select(c => c.ToString()).ToList();
+            foreach(var cmdlet in functionCmdlets)
+            {
+                var metadata = new CmdletMetadata();
+                metadata.VerbName = cmdlet.Split("-")[0];
+                metadata.NounName = cmdlet.Split("-")[1];
+                allCmdlets.Add(metadata);
             }
 
+            ValidateHelpRecords(allCmdlets, helpFiles, helpLogger);
+            ValidateHelpMarkdown(helpFolder, helpFiles, helpLogger);
+
             Directory.SetCurrentDirectory(savedDirectory);
             
         }
 
         private void ValidateHelpRecords(IList<CmdletMetadata> cmdlets, IList<string> helpRecords,
             ReportLogger<HelpIssue> helpLogger)
         {
+            var cmdletDict = new Dictionary<string, CmdletMetadata>();
             foreach (var cmdlet in cmdlets)
             {
+                cmdletDict.Add(cmdlet.Name, cmdlet);
                 if (!helpRecords.Contains(cmdlet.Name, StringComparer.OrdinalIgnoreCase))
                 {
-                    helpLogger.LogRecord(new HelpIssue
+                    HelpIssue issue = new HelpIssue
                     {
                         Target = cmdlet.ClassName,
                         Severity = 1,
                         ProblemId = MissingHelp,
-                        Description = string.Format("Help missing for cmdlet {0} implemented by class {1}",
-                        cmdlet.Name, cmdlet.ClassName),
                         Remediation = string.Format("Add Help record for cmdlet {0} to help file.", cmdlet.Name)
-                    });
+                    };
+                    if(cmdlet.ClassName != null)
+                    {
+                        issue.Description = $"Help missing for cmdlet {cmdlet.Name} implemented by class {cmdlet.ClassName}";
+                    }
+                    else
+                    {
+                        issue.Description = $"Help missing for cmdlet {cmdlet.Name} implemented by functions";
+                    }
+                    helpLogger.LogRecord(issue);
+                }
+            }
+
+            foreach (var helpRecord in helpRecords)
+            {
+                if(!cmdletDict.ContainsKey(helpRecord))
+                {
+                    Console.Error.WriteLine($"Help record {helpRecord} has no cmdlet.");
+                } 
+            }
+        }
+
+        private void ValidateHelpMarkdown(string helpFolder, IList<string> helpRecords, ReportLogger<HelpIssue> helpLogger)
+        {
+            foreach(var helpMarkdown in helpRecords)
+            {
+                var file = Directory.GetFiles(helpFolder, helpMarkdown + ".md").FirstOrDefault();
```

Comment:
Suggest to use Path.Combine(helpFolder, helpMarkdown + ".md") to get file full path directly